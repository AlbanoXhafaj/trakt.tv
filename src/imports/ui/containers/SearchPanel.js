import React from "react";
import PropTypes from 'prop-types';

// OnChange pass the text of input as parameter
// Input box value equals to the search filters text, needed for wipe out by other actions
const SearchPanel = ( { onChange, text }) => (
  <div className="inline">
    <span>Search:</span>
    <input
      type="text"
      placeholder="keyword..."
      value = { text }
      onChange = { (e) => onChange(e.target.value) }
    />
  </div>
);

SearchPanel.propTypes = {
  onChange: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired
};

export default SearchPanel;
