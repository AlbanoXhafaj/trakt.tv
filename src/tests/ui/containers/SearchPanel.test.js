import "../../setups/setup_enzyme.js";
import React from "react";

import { shallow , mount } from "enzyme";
import toJSON from 'enzyme-to-json';

import SearchPanel from "containers/SearchPanel";

describe("<SearchPanel  />", () => {
  const onChange = jest.fn();
  const initialProps = {
      text: "search text",
      onChange: onChange
  };

  let wrapper = shallow(<SearchPanel {...initialProps} />);

  it("should match with the snapshot" , () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should pass the value of the input as an argument on tha change event",()=> {
    const event = {
      preventDefault(){},
      target: { value: initialProps.text }
    }

    wrapper.find("input").simulate('change', event);
    expect(onChange).toBeCalledWith(initialProps.text);
  })
});
