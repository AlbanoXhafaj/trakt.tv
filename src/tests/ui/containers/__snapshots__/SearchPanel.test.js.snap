// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchPanel  /> should match with the snapshot 1`] = `
<div
  className="inline"
>
  <span>
    Search:
  </span>
  <input
    onChange={[Function]}
    placeholder="keyword..."
    type="text"
    value="search text"
  />
</div>
`;
